### PR TITLE
Fix typos and small issues with the tutorial

### LINKED ISSUE #122: Data location is ambiguous
It is unclear where the data has to go. I'm following the tutorial `Using your own machine`

The wget script downloads the data to `$HOME/default_inputpath/`:

```bash
wget --no-clobber --input-file \
  https://github.com/ESMValGroup/ESMValTool_Tutorial/raw/master/data/dataset.urls \
  --directory-prefix $HOME/default_inputpath/
```

But in the setup, we are asked to use `~/esmvaltool_tutorial/data`:

```yaml
  rootpath:
  ...
    CMIP5: [~/cmip5_inputpath1, ~/cmip5_inputpath2, ~/esmvaltool_tutorial/data]
```

However, this results in `esmvalcore._recipe_checks.RecipeError: Missing data`.

### LINKED ISSUE #123: Missing data in dataset.urls
Some datasets are missing from https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/master/data/dataset.urls:

```
http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MOHC/HadGEM2-ES/historical/mon/atmos/Amon/r1i1p1/v20120928/tas/tas_Amon_HadGEM2-ES_historical_r1i1p1_193412-195911.nc
http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MOHC/HadGEM2-ES/historical/mon/atmos/Amon/r1i1p1/v20120928/ts/ts_Amon_HadGEM2-ES_historical_r1i1p1_193412-195911.nc
```

This causes the following error in the tutorial (https://esmvalgroup.github.io/ESMValTool_Tutorial/04-recipe/index.html):
```esmvalcore._recipe_checks.RecipeError: No input data available for years 1935, 1936, 1937, 1938, 1939, 1940, 1941, 1942, 1943, 1944, 1945, 1946, 1947, 1948, 1949, 1950, 1951, 1952, 1953, 1954, 1955, 1956, 1957, 1958 in files ['/home/stef/default_inputpath/tas_Amon_HadGEM2-ES_historical_r1i1p1_198412-200511.nc', '/home/stef/default_inputpath/tas_Amon_HadGEM2-ES_historical_r1i1p1_195912-198411.nc', '/home/stef/default_inputpath/tas_Amon_HadGEM2-ES_historical_r1i1p1_188412-190911.nc', '/home/stef/default_inputpath/tas_Amon_HadGEM2-ES_historical_r1i1p1_190912-193411.nc']```

### PR DESCRIPTION
**Pull request by stefsmeets**
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [ ] Give this pull request a descriptive title.
- [ ] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [ ] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [ ] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Hi guys, just creating this pull request to keep track of some small issues I'm running into with the tutorial. It's mostly typos, formatting errors, and some clarifications here and there.

- Closes #123 
- Closes #122

### CODE REVIEW

File: _episodes/02-installation.md [hunk 1]
```diff
@@ -40,6 +40,10 @@ documentation](https://docs.conda.io/projects/conda/en/latest/user-guide/install
     should work on any recent system. If you have problems in the next step(s)
     you can alternatively try a 32 bit version.
```
Comment (SarahAlidoost) [inline _episodes/02-installation.md:41]: Please replace all of the text before the code block below with "Please download Miniconda3 by running"

File: _episodes/02-installation.md [hunk 2]
```diff
@@ -135,10 +139,10 @@ Next, download the file
 [`julia-1.0.5-linux-x86_64.tar.gz`](https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.5-linux-x86_64.tar.gz)
 by clicking the link or going to the [Julia downloads
 page](https://julialang.org/downloads/). Download the file directly to the
-`~/julia` directory or move it there after downloading. To extract the file, you
-can use the following command:
+`~/julia` directory or move it there after downloading. To download and extract the file, you can use the following commands:
```
Comment (SarahAlidoost) [inline _episodes/02-installation.md:142]:
```suggestion
Next, to download and extract the file `julia-1.0.5-linux-x86_64.tar.gz`, you can use the following commands:
```

File: _episodes/02-installation.md [hunk 3]
```diff
@@ -135,10 +139,10 @@ Next, download the file
 [`julia-1.0.5-linux-x86_64.tar.gz`](https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.5-linux-x86_64.tar.gz)
 by clicking the link or going to the [Julia downloads
 page](https://julialang.org/downloads/). Download the file directly to the
```
Comment (SarahAlidoost) [inline _episodes/02-installation.md:141]: Please remove this paragraph, no need to have three different instructions for downloading the file.

File: _episodes/04-recipe.md [hunk 4]
```diff
@@ -398,16 +398,20 @@ Exemplary output (depending on the directory paths and package versions that are
 available) can be found below. Note that the timestamps differ.
 
 > ## Your output plot(s).
-> Plot for the dataset(s):
->
+> Plots for the dataset(s) are located in `./recipe_example_#_#/diag_timeseries_temperature/timeseries_diag/`
```
Comment (SarahAlidoost) [inline _episodes/04-recipe.md:401]:
```suggestion
> Plots for the dataset(s) are located in `./recipe_example_#_#/plots/diag_timeseries_temperature/timeseries_diag/`
```

File: _episodes/06-debugging.md [hunk 5]
```diff
@@ -231,7 +231,7 @@ config_developer_file: {HOME}/.esmvaltool/config_reference.yml
 > ## ESMValTool can’t locate the data
 >
 > You are assisting a colleague with ESMValTool. The colleague changes the
```
Comment (SarahAlidoost) [inline _episodes/06-debugging.md:233]:
```suggestion
> You are assisting a colleague with ESMValTool. The colleague replaces the
```

File: _episodes/06-debugging.md [hunk 6]
```diff
@@ -231,7 +231,7 @@ config_developer_file: {HOME}/.esmvaltool/config_reference.yml
...
-> ``project`` entry to ``CMIP6`` and runs the recipe. However, ESMValTool
+> ``datasets/project`` entry to ``CMIP6`` and runs the recipe. However, ESMValTool
```
Comment (SarahAlidoost) [inline _episodes/06-debugging.md:234]:
```suggestion
> "CMIP5" in ``project: CMIP5`` to ``CMIP6`` and runs the recipe. However, ESMValTool
```

File: _episodes/06-debugging.md [hunk 7]
```diff
@@ -300,20 +300,28 @@ esmvalcore._task.DiagnosticError: Cannot execute script 'diag_scripts/examples/d
...
-recipe:
+``<path_to_esmvaltool>/diag_scripts/ocean/``. Alternatively, the script path can
+be an absolute path. To examine this, we create a small python script `hello.py`:
+
+```python
+with open('hello_world.txt', 'w') as f:
+    print('Fello from ESMValTool!', file=f)
+```
+
+Change and run the recipe:
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/06-debugging.md:311]: Nice example. However, this section aims to show the difference between the absolute and relative path of the diagnostic script. "creating your own diagnostic" is another episode in the advanced part of the tutorial that would be developed in the near future.

File: _episodes/06-debugging.md [hunk 8]
```diff
@@ -300,20 +300,28 @@ esmvalcore._task.DiagnosticError: Cannot execute script 'diag_scripts/examples/d
...
 
 ```yaml
 40        scripts:
 41          timeseries_diag:
-42            script: path_to_esmvaltool/diag_scripts/ocean/diagnostic_timeseries.py
+42            script: ~/esmvaltool_tutorial/hello.py
```
Comment (SarahAlidoost) [inline _episodes/06-debugging.md:316]:
```suggestion
42            script: <path_to_esmvaltool>/diag_scripts/ocean/diagnostic_timeseries.py
```

File: _episodes/06-debugging.md [hunk 9]
```diff
@@ -300,20 +300,28 @@ esmvalcore._task.DiagnosticError: Cannot execute script 'diag_scripts/examples/d
...
 ```
 
 ~~~bash
   esmvaltool run recipe_example.yml
 ~~~
 
+Now examine `./esmvaltool_output/recipe_example_#_#/run/diag_timeseries_temperature/timeseries_diag/` to see if it worked! Of course, this is a simple example, but it shows how you can customize diagnostic scripts or even make your own. Have a look through the scripts available in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/diag_scripts/).
+
```
Comment (SarahAlidoost) [inline _episodes/06-debugging.md:324]:
```suggestion
Now examine `./esmvaltool_output/recipe_example_#_#/run/diag_timeseries_temperature/timeseries_diag/` to see if it worked! Of course, this is a simple example, but it shows how you can customize diagnostic scripts or even make your own.

```

Comment (SarahAlidoost) [reply _episodes/06-debugging.md:324]: "creating your own diagnostic" is another episode in the advanced part of the tutorial that would be developed in the near future. Also, "Available recipe and diagnostic scripts" are already mentioned in the next section.

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@stefsmeets thank you for the improvements. Please see my comments.

File: _episodes/06-debugging.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (stefsmeets) [reply _episodes/06-debugging.md:311]: It was the only thing I could get to work with my limited knowledge 3 weeks ago 😅

The absolute path you are referring to here is very deep in miniconda, so I didn't think the instruction was very useful. The path for me is nested somewhere in: `~/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/`. I think it is bad practice to recommend people to go looking around there. What do you think about adding a link to the diag script from the ESMValTool directory and running them from the working directory?

**Comment by stefsmeets** [comment]
I addressed your comments. There is just one part that I don't find very satisfying. Please have a look at my comment here:
https://github.com/ESMValGroup/ESMValTool_Tutorial/pull/125#discussion_r483464828

I reverted my changes, but I think this part needs some more thought.

**Review by SarahAlidoost — APPROVED** [review]
@stefsmeets thank you.

**Pull request merged**